addons/google_calendar/models/google_calendar.py

```diff
@@ -225,12 +225,12 @@ def generate_data(self, event, isCreating=False):
             "start": {
                 type: start_date,
                 vstype: None,
-                'timeZone': self.env.context.get('tz') or 'UTC',
+                'timeZone': self.env.context.get('tz') or self.env.user.partner_id.tz or 'UTC',
```

Comment:
Minor detail: `self.env.user.tz` is enough since `res.users` inherits from `res.partner`:

https://github.com/odoo/odoo/blob/2617f8c27262d8c923bea376b64d04e1804da15b/odoo/addons/base/models/res_users.py#L202